src/util/parse_color.ts

```diff
@@ -24,14 +28,33 @@ import {HSLColor, hslToRgb, RGBColor} from './color_spaces';
  *   - legacy color syntax rgba() is supported with an identical grammar and behavior to rgb()
  *   - legacy color syntax hsla() is supported with an identical grammar and behavior to hsl()
  *
- * @param input CSS color string to parse.
+ * @param input Color string to parse.
  * @returns Color in sRGB color space, with `red`, `green`, `blue`
  * and `alpha` channels normalized to the range 0..1,
  * or `undefined` if the input is not a valid color string.
  */
-export function parseCssColor(input: string): RGBColor | undefined {
+export function parseColor(input: string): RGBColor | undefined {
     input = input.toLowerCase();
+    return input.startsWith('[') ? parseJsonArrayColor(input) : parseCssColor(input);
+}
+
+function parseJsonArrayColor(input: string): RGBColor | undefined {
+    try {
+        const [r, g, b, a] = JSON.parse(input);
+        if (!isNaN(r + g + b)) {
+            return [
+                clamp(+r / 255, 0, 1),
+                clamp(+g / 255, 0, 1),
+                clamp(+b / 255, 0, 1),
+                clamp(isNaN(a) ? 1 : +a, 0, 1)];
+        }
+    } catch(ex) {
```

Comment:
My understanding previously was that try/catch was slower. But looks like modern browsers have improved and now I don't see any perf issue with using try/catch. Adding try/catch here allow to catch all invalid JSON string.

https://jsbench.me/mbli72txix
![image](https://github.com/maplibre/maplibre-style-spec/assets/15951646/26bf804f-b5bd-4008-bf27-65036098dbf5)

Reply:
I think try-catch is slower compared to ifs, in case you go through the catch part of the try-catch, at least is what I was taught long time ago... :-)

Reply:
Yes, if code goes inside catch then its quite slow. But for any production style, it should never go into catch block as it will be a invalid color and thus production code does not suffer from any performance degradation in this case. Overhead of adding various checks to validate JSON format will eat away any performance benefits apart from adding code complications.

https://jsbench.me/mbli72txix/2
![image](https://github.com/maplibre/maplibre-style-spec/assets/15951646/ab9d3d2c-b40f-4420-8f72-b05f757289db)